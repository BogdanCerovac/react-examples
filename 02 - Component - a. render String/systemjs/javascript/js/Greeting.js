/*! Mozilla Public License Version 2.0 !*/
/*! Copyright Â© 2018 Rick Beerendonk   !*/

function Greeting() {
  return 'Hello World!';
}
export default Greeting;
