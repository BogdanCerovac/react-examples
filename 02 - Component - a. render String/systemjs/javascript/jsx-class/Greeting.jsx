/*! Mozilla Public License Version 2.0 !*/
/*! Copyright Â© 2018 Rick Beerendonk   !*/

import React from 'react';

class Greeting extends React.Component {
  render() {
    return 'Hello World!';
  }
}

export default Greeting;
