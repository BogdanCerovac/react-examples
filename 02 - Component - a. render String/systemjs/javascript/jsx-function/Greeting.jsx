/*! Mozilla Public License Version 2.0 !*/
/*! Copyright Â© 2018 Rick Beerendonk   !*/

const Greeting = () => 'Hello World!';

export default Greeting;
