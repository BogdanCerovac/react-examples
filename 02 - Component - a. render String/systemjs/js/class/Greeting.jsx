/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2018 Rick Beerendonk          !*/

import React from 'react';

class Greeting extends React.Component {
  render() {
    return 'Hello World!';
  }
}

export default Greeting;
