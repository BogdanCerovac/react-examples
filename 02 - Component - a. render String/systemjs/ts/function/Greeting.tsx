/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2018 Rick Beerendonk          !*/

function Greeting(): string {
  return 'Hello World!';
}

export default Greeting;
