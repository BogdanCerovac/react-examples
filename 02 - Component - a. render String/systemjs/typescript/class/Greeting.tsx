/*! Mozilla Public License Version 2.0 !*/
/*! Copyright Â© 2018 Rick Beerendonk   !*/

import React from 'react';

class Greeting extends React.Component<undefined, undefined> {
  render(): string {
    return 'Hello World!';
  }
}

export default Greeting;
