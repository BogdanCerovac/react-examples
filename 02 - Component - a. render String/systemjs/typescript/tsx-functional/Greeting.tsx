/*! Mozilla Public License Version 2.0 !*/
/*! Copyright Â© 2018 Rick Beerendonk   !*/

import React from 'react';

const Greeting = (): string => 'Hello World!';

export default Greeting;
