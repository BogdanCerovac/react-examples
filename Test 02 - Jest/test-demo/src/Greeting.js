import React from 'react';

/* eslint react/prop-types: "off" */

class Greeting extends React.Component {
  constructor(props) {
    super(props);

    if (!props.name) {
      throw new Error('No name provided');
    }
  }
  render() {
    return <h1>Hello {this.props.name}!</h1>;
  }
}

export default Greeting;
