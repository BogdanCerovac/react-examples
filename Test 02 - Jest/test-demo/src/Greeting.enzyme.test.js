import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Greeting from './Greeting';

/* eslint-env jest */

/// Enzyme ///

it('renders without crashing (enzyme)', () => {
  shallow(<Greeting />);
});

it('renders hello world message (enzyme)', () => {
  const wrapper = shallow(<Greeting />);

  expect(wrapper.type()).toEqual('h1');
  expect(wrapper.children().length).toEqual(1);
  expect(wrapper.childAt(0).node).toEqual('Hello World!');
  
  expect(wrapper.html()).toEqual('<h1>Hello World!</h1>');
});

it('renders hello world snapshot (enzyme)', () => {
  const wrapper = shallow(<Greeting />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
