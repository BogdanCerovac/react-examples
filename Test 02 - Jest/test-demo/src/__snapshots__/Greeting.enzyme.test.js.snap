// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders hello world snapshot (enzyme) 1`] = `
<h1>
  Hello World!
</h1>
`;
